rewriter/rewriter.cc

```diff
@@ -122,6 +124,11 @@ class Rewriterer {
                         replaceNodes[stat.get()] = std::move(nodes);
                         return;
                     }
+
+                    // Determines if there is an `extend ActiveSupport::Concern`. Used later.
+                    if (Concern::doesExtendConcern(ctx, send)) {
+                        extendsConcern = true;
+                    }
```

Comment:
This check can be inserted inside the rewriter and let the rewriter traverse the tree as well. We thought this way would be a good performance improvement however, I don't know how much it actually affects performance. I wanted to point it out since this is unique to this rewriter.

Reply:
Yeah I'm not too worried about this for performance. Would you mind moving this to be the first thing that `Concern::run` does? (Given that it's already operating on the whole ClassDef it shouldn't be too much work).

Reply:
This is now triggered within `Concern::run`. I repeated the usage of `typecase()` switch statement similar to `rewriter.cc`. If there is a simpler approach you recommend I can change it.